subgraphs/portal/src/mappings/handlers.ts

```diff
@@ -0,0 +1,478 @@
+  getTokenParams(address: Address): TokenParams {
+    const erc20 = _ERC20.bind(address);
+    const name = erc20.name();
+    const symbol = erc20.symbol();
+    const decimals = erc20.decimals().toI32();
+
+    return {
+      name,
+      symbol,
+      decimals,
+    };
+  }
+}
+
+export function handleSwapOut(event: LogMessagePublished): void {
+  const payload = event.params.payload;
+
+  let amount: BigInt;
+  let tokenAddress: Bytes;
+  let tokenChain: i32;
+  let to: Bytes;
+  let toChain: i32;
+
+  const bridgeContract = Bridge.bind(NetworkConfigs.getBridgeAddress());
+  const parseTransferCall = bridgeContract.try_parseTransfer(payload);
+  if (parseTransferCall.reverted) {
+    const parseTransferWithPayloadCall =
+      bridgeContract.try_parseTransferWithPayload(payload);
+    if (parseTransferWithPayloadCall.reverted) {
+      log.warning("[handleSwapOut] tx_hash: {} failed to parse payload: {}", [
+        event.transaction.hash.toHexString(),
+        payload.toHexString(),
+      ]);
+      return;
+    }
+
+    const transferStruct = parseTransferWithPayloadCall.value;
+
+    amount = transferStruct.amount;
+    tokenAddress = transferStruct.tokenAddress;
+    tokenChain = transferStruct.tokenChain;
+    to = transferStruct.to;
+    toChain = transferStruct.toChain;
+  } else {
+    const transferStruct = parseTransferCall.value;
+
+    amount = transferStruct.amount;
+    tokenAddress = transferStruct.tokenAddress;
+    tokenChain = transferStruct.tokenChain;
+    to = transferStruct.to;
+    toChain = transferStruct.toChain;
+  }
+
+  const chainID = NetworkConfigs.getNetworkID();
+
+  const sdk = SDK.initialize(conf, new Pricer(), new TokenInit(), event);
+
+  let bridgePoolType = BridgePoolType.LOCK_RELEASE;
+  let crosschainTokenType = CrosschainTokenType.WRAPPED;
+  if (BigInt.fromI32(tokenChain) != chainID) {
+    tokenAddress = bridgeContract.wrappedAsset(tokenChain, tokenAddress);
+    bridgePoolType = BridgePoolType.BURN_MINT;
+    crosschainTokenType = CrosschainTokenType.CANONICAL;
+  }
+
+  if (tokenAddress == Address.fromString(ZERO_ADDRESS)) {
+    return;
+  }
+
+  const token = sdk.Tokens.getOrCreateToken(truncateAddress(tokenAddress));
+  amount = denormalizeAmount(amount, token.decimals);
+
+  const pool = sdk.Pools.loadPool<string>(token.id);
+  if (!pool.isInitialized) {
+    pool.initialize(token.name, token.symbol, bridgePoolType, token);
+  }
+
+  const crosschainTokenAddr = getCrossTokenAddress(
+    token.id.toHexString(),
+    BigInt.fromI32(tokenChain),
+    chainID,
+    BigInt.fromI32(toChain)
+  );
+  const crosschainToken = sdk.Tokens.getOrCreateCrosschainToken(
+    BigInt.fromI32(toChain),
+    Bytes.fromUTF8(crosschainTokenAddr),
+    crosschainTokenType,
+    Address.fromBytes(token.id)
+  );
+
+  pool.addDestinationToken(crosschainToken);
+
+  const route = pool.getDestinationTokenRoute(crosschainToken);
+
+  const coreContract = Core.bind(dataSource.address());
```

Comment:
Can we store this as a helper field in `Protocol`? Is there an event that is emitted each time this is updated to reduce the number of times we need to call this function?

Reply:
added helper field in `Protocol`.
and, no there is no event for this fee update [ref](https://etherscan.io/address/0x3c3d457f1522d3540ab3325aa5f1864e34cba9d0#code#F6#L54)